pkg/cedar/cedarldap/translated_client.go

```diff
@@ -107,12 +108,19 @@ func (c TranslatedClient) FetchUserInfos(ctx context.Context, euaIDs []string) (
 		}
 	}
 
-	idsStr := strings.Join(euaIDs, ",")
+	// this function's caller might not expect euaIDs to be mutated by sorting,
```

Comment:
@DylanSpOddball I am just being a nerd, but this would be a fun spot to use a merge sort and save a negligible (O(log n)) amount of space. You would be using O(n) space with a more common implementation of merge sort that doesn't sort in place (avoiding side effects on the input). Compared to the Go sort algorithm, which uses O(log n) space for their "in-place" (relaxed definition) sort, on top of the O(n) you're using to avoid side effects.

There is some discussion on the [Go sort repo here](https://github.com/golang/go/blob/c847a2c9f024f47eee25c132f2d80e7037adea36/src/sort/sort.go#L176) if you are as bored as I am right now :). They're doing a variant of a merge sort and using a relaxed definition of "in-place" since it uses O(log n) space.

Reply:
Interesting! I don't think this code needs to worry about the tiny bit of extra memory by using Go's sorting functions, but it's interesting to read about and good to keep in mind. Feel free to bring up interesting info like this, especially theoretical CS stuff. 😃